test/compress/issue-1443.js

```diff
@@ -0,0 +1,53 @@
+unsafe_undefined: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {}
+    input: {
+        function f(undefined) {
+            var a;
+            return function() {
+                console.log(typeof a == "undefined");
+            }
+        }
+    }
+    expect: {
+        function f(n) {
+            var o;
+            return function() {
+                console.log(n===o);
+            }
+        }
+    }
+}
+
+keep_fnames: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {
+        keep_fnames: true
+    }
+    input: {
+        function f(undefined) {
+            return function() {
+                function n(a) {
+                    return a * a;
+                }
+                return typeof n == "undefined";
+            };
+        }
+    }
+    expect: {
+        function f(r) {
+            return function() {
+                function n(n) {
+                    return n * n;
+                }
+                return r === n;
```

Comment:
How does it arrive at this result?

`typeof n` is `"function"`

`"function" == "undefined"` is `false`